app/graphql/mutations/create_sample.rb

```diff
@@ -8,23 +8,32 @@ class CreateSample < BaseMutation
     argument :description, String, description: 'The description to give the sample.'
     argument :name, String, required: true, description: 'The name to give the sample.'
     argument :project_id, ID, # rubocop:disable GraphQL/ExtractInputType
-             required: true,
-             description: 'The Node ID of the project to switch the sample will be created in.'
+             required: false,
+             description: 'The Node ID of the project. For example, `gid://irida/Project/a84cd757-dedb-4c64-8b01-097020163077`.' # rubocop:disable Layout/LineLength
+    argument :project_puid, ID, # rubocop:disable GraphQL/ExtractInputType
+             required: false,
+             description: 'Persistent Unique Identifier of the project. For example, `INXT_PRJ_AAAAAAAAAA`.'
+    validates required: { one_of: %i[project_id project_puid] }
 
     field :errors, [String], null: false, description: 'A list of errors that prevented the mutation.'
     field :sample, Types::SampleType, description: 'The newly created sample.'
 
-    def resolve(name:, description:, project_id:)
-      project = IridaSchema.object_from_id(project_id, { expected_type: Project })
-      sample = Samples::CreateService.new(current_user, project, { name:, description: }).execute
+    def resolve(args) # rubocop:disable Metrics/MethodLength
+      project = if args[:project_id]
+                  IridaSchema.object_from_id(args[:project_id], { expected_type: Project })
+                else
+                  Project.find(args[:project_puid])
```

Comment:
Same comment as above about using `find_by` instead of `find`

Reply:
Fixed in c362e774 good catch!